spec/MongoStorageAdapter.spec.js

```diff
@@ -316,4 +319,234 @@ describe_only_db('mongo')('MongoStorageAdapter', () => {
+          .adapter;
+        spyOn(
+          databaseAdapter.database.serverConfig,
+          'command'
+        ).and.callThrough();
+
+        await request({
+          method: 'POST',
+          headers: headers,
+          url: 'http://localhost:8378/1/batch',
+          body: JSON.stringify({
+            requests: [
+              {
+                method: 'PUT',
+                path: '/1/classes/MyObject/' + myObject.id,
+                body: { myAttribute: 'myValue' },
+              },
+            ],
+          }),
+        });
+
+        let found = false;
+        databaseAdapter.database.serverConfig.command.calls
+          .all()
+          .forEach(call => {
+            found = true;
+            expect(call.args[2].session).toBe(undefined);
+          });
+        expect(found).toBe(true);
+      });
+
+      it('should not use transaction in a put request', async () => {
+        const myObject = new Parse.Object('MyObject');
+        await myObject.save();
+
+        const databaseAdapter = Config.get(Parse.applicationId).database
+          .adapter;
+        spyOn(
+          databaseAdapter.database.serverConfig,
+          'command'
+        ).and.callThrough();
+
+        await request({
+          method: 'PUT',
+          headers: headers,
+          url: 'http://localhost:8378/1/classes/MyObject/' + myObject.id,
+          body: { myAttribute: 'myValue' },
+        });
+
+        let found = false;
+        databaseAdapter.database.serverConfig.command.calls
+          .all()
+          .forEach(call => {
+            found = true;
+            expect(call.args[2].session).toBe(undefined);
+          });
+        expect(found).toBe(true);
+      });
+
+      it('should not use transaction when using the sdk', async () => {
+        const databaseAdapter = Config.get(Parse.applicationId).database
+          .adapter;
+        spyOn(
+          databaseAdapter.database.serverConfig,
+          'command'
+        ).and.callThrough();
+        spyOn(
+          databaseAdapter.database.serverConfig,
+          'cursor'
+        ).and.callThrough();
+        spyOn(
+          databaseAdapter.database.serverConfig,
+          'update'
+        ).and.callThrough();
+        spyOn(
+          databaseAdapter.database.serverConfig,
+          'insert'
+        ).and.callThrough();
+        spyOn(
+          databaseAdapter.database.serverConfig,
+          'remove'
+        ).and.callThrough();
+
+        const myObject = new Parse.Object('MyObject');
+        await myObject.save();
+
+        myObject.set('myAttribute', 'myValue');
+        await myObject.save();
+
+        let found = false;
+        databaseAdapter.database.serverConfig.command.calls
+          .all()
+          .forEach(call => {
+            found = true;
+            expect(call.args[2].session).toBe(undefined);
+          });
+        expect(found).toBe(true);
+        databaseAdapter.database.serverConfig.cursor.calls
```

Comment:
Including an assertion of the number of calls would be a good idea, since you're not using `found`.

Reply:
I am actually not sure that the other operations will be called. The right one is the one with the found. I added the others just in case.

Reply:
I took the freedom to commit the separation of these tests. What do you think?